src/openads-appnexus/repository/AdListenerRepository.js

```diff
@@ -0,0 +1,51 @@
+import ReplayEventBus from '../service/ReplayEventBus'
+
+const AD_STORED = 'AD_STORED'
+
+const DEFAULT_TIMEOUT = 20000
+const TIMEOUT_EXCEPTION = 'Timeout waiting to retrieve the Ad'
+
+export default class AdListenerRepository {
+  constructor({timeout = DEFAULT_TIMEOUT} = {}) {
+    this._timeout = timeout
+  }
+
+  find({id}) {
+    return Promise.race([
+      new Promise((resolve, reject) => {
+        const eventId = this._eventId({id})
```

Comment:
I would use _evenName_ instead of _eventId_ and named the method with a self explained name:

```javascript
const eventName = this._generateEventNameFromEventId({id})
```